app/models/miq_server/role_management.rb

```diff
@@ -46,6 +46,7 @@ def set_active_role_flags
     self.has_active_userinterface = self.has_active_role?("user_interface")
     self.has_active_websocket     = self.has_active_role?("websocket")
     self.has_active_webservices   = self.has_active_role?("web_services")
+    self.has_active_cockpit_ws    = MiqCockpitWsWorker.can_activate_cockpit?
```

Comment:
This is mainly to cover non-appliance use-cases, ei dev machines or also because of the someone disjoint release schedules to ensure that things wouldn't break if this was installed on a machine that did not yet have a compatible version of cockpit available.